test/functional/test_framework/test_node.py

```diff
@@ -235,7 +235,10 @@ def stop_node(self, expected_stderr='', wait=0):
         try:
             self.drain_main_signal_callbacks_pending()
             self.stop(wait=wait)
-        except (http.client.CannotSendRequest, subprocess.CalledProcessError):
+        except (ConnectionError, http.client.HTTPException, subprocess.CalledProcessError):
+            # Most likely, the node is already stopped. Print the exception and continue.
+            # If we want to be sure of that, we should use TestFramework.stop_node, instead
+            # of TestNode.stop_node directly, since the former asserts that the process has ended.
             self.log.exception("Unable to stop node.")
```

Comment:
I'm seeing that once we call `stop_node`, we don't reset `self.running` to False at the end or in any other place of this method. Might this help?

Reply:
`self.running` is set once the process is no longer running -- and, at the end of this method, it's still active (but in the process of shutting itself down, flushing the wallet etc.)
If the test wants to make sure that the node has stopped, it must run `TestNode.wait_until_stopped()` -- and then `self.running` will be set to `False`.